draft-ietf-mls-architecture.md

```diff
@@ -388,31 +388,33 @@ controller who can modify the group. MLS is compatible with having group
 restrictions are enforced by the application layer.
 
-## Group Members and Clients
-
-While informally, a group can be considered to be a set of users possibly using
-multiple endpoint devices to interact with the Service Provider, this definition
-is too simplistic.
-
-Formally, a client is a set of cryptographic objects composed of public values
-such as a name (an identity), a public encryption key, and a public signature
-key. Ownership of a client by a user is determined by the fact that the user has
-knowledge of the associated secret values. When a client is part of a Group, it
-is called a Member.
-In some messaging systems, clients belonging to the same user must all share the
-same signature key pair, but MLS does not assume this.
-
-Users will often use multiple devices, e.g., a phone as well as a laptop.
-Different devices may be represented as different clients, with independent
-cryptographic state.
-The formal definition of a Group in MLS is the set of clients that have
-knowledge of the shared group secret established in the group key establishment
-phase of the protocol and have contributed to it.
-Until a Member has been added to the group and contributed to the group secret
-in a manner verifiable by other members of the group, other members cannot
-assume that the Member is a member of the group.
-Different devices are represented as different clients with independent
-cryptographic state.
+## Users, Clients, and Groups
+
+While it's natural to think of a messaging system as consisting of groups
+of users, possibly using different devices, in MLS the basic unit of
+operation is not the user but rathr the "client".
+Formally, a client is a set of cryptographic objects composed of
+public values such as a name (an identity), a public encryption key,
+and a public signature key. As usual, a user demonstrates ownership of
+the client by demonstrating knowledge of the associated secret
+values.
+
+In some messaging systems, clients belonging to the same user
+must all share the same signature key pair, but MLS does not assume
+this; instead a user may have multiple clients with the same identity
+and different keys. In this case, each client will have its own
+cryptographic state, and it is up to the application to determine
+how to present this situation to users. For instance, it may render
+messages to and from a given user identically regardless of which
+client they are associated with, or may choose to distinguish them.
+
+When a client is part of a Group, it is called a Member.  A group in
+MLS is defined as the set of clients that have knowledge of the shared
+group secret established in the group key establishment phase of the
+protocol and have contributed to it.  Until a Member has been added to
+the group and contributed to the group secret in a manner verifiable
+by other members of the group, other members cannot assume that the
+Member is a member of the group.
```

Comment:
@bifurcation I'm not sure how helpful this last sentence is (it is in the original). I suppose the idea here is that (for instance) the Welcome message could be corrupt or the new member has lost their key. However, it seems like as a practical matter, when Bob learns that Charlie has been added to the group, then it is reasonable for him to assume that he is.

After all, Charlie could lose his key *after* being added to the group. Thoughts?

Reply:
I am gonna slightly change the sentence, but there were concerns in the past about not having this. The idea was that anyone can be added to a group without their content. So we want to consider clients to be members only when the Welcome information was processed and the client has made an update.

Reply:
I think the problem here is that the phrase "member of the group" does not have an intuitive meaning

Like in this case, Charlie (1) is getting copies of the message and (2) can read them, so how is he not a member of the group

Isn't this just the standard two generals stuff.

Reply:
Well the definition we used until now is that it is not a group member until it has contributed to the epoch secret.
This is not amazing, I agree, but this is the current status-quo... What we wanted to avoid is that if I take one of your KeyPackages and add you to "the evil group" you are automatically considered a member of the group.